src/app/(public-pages)/deals/add/actions.tsx

```diff
@@ -30,24 +31,32 @@ export const submitDealAction = async (
 
   if (validated.success) {
     try {
-      await createDeal(validated.data)
+      const deal = await createDeal(validated.data)
+      try {
```

Comment:
Can we separate these try/catch into two separate ones instead of nesting them?

Reply:
I'm a little stuck on this. This was one of the reasons I had the separate function initially, this is the best way I can separate everything so it isn't nested.

```js
if (!validated.success) {
  return {
    success: false,
    message: 'Validation failed',
  }
}

let deal;
try {
  deal = await createDeal(validated.data)
} catch (error) {
  console.error('Error in deal creation process:', error)
  return {
    success: false,
    message: 'There was an error creating the deal',
  }
}

try {
  const { ids } = await inngest.send({
    name: 'admin/new-deal-created',
    data: {
      dealId: deal.xata_id,
    },
  })
} catch (error) {
  console.error('Error sending emails:', error)
}
```

I thought it was more readable with the separate function to encapsulate some of the logic and label it with a function name.  Is the code above what you are looking for? I'm good with either approach, I'm curious if there is a better way to separate everything into individual `try/catch`.

Reply:
I think this works. The other alternative is to have one try/catch that handles specific errors. So, if you receive an Inngest error you could just log the error and not return. If it's anything else you could log and return. In the code snippet above, you'll still need to return the success case after the second try/catch

Reply:
Have you looked at any specific errors returned by Inngest?

Reply:
I think this should do it, thanks for the advice on handling all errors in the catch and filtering them:

```js
if (validated.success) {
    try {
      const deal = await createDeal(validated.data)
      await inngest.send({
        name: 'admin/new-deal-created',
        data: {
          dealId: deal.xata_id,
        },
      })
      return { success: true }
    } catch (error) {
      if (error instanceof StepError) {
        console.error('Error sending emails:', error)
      } else {
        console.error('Error in deal creation process:', error)
        return {
          success: false,
          message: 'There was an error creating the deal',
        }
      }
    }
  }
  // Handle validation failure
  const errors = validated.error?.issues.reduce((acc: FormErrors, issue) => {
    const path = issue.path[0] as string
    acc[path] = issue.message
    return acc
  }, {})
  return { success: false, errors }
  ```
  
There were a few spots where I didn't wrap step functions in a `try/catch` (the `try/catch` were within the step function). I refactored those so the instance will be of `StepError`. Here is a link to the [StepError](https://www.inngest.com/docs/features/inngest-functions/error-retries/inngest-errors#step-errors) docs. I wish there were a more generic inngest error, like `InngestError extends Error` and then all of the Inngest errors extend from `InngestError` but it looks like all of the error types listed in the docs extend from standard `Error`.